lib/jobs/ScheduleManager.js

```diff
@@ -3,32 +3,48 @@
 const _ = require('lodash');
 const Promise = require('bluebird');
 const pubsub = require('pubsub-js');
+var moment = require('moment-timezone');
 const utils = require('../utils');
 const config = require('../config');
 const Scheduler = require('./Scheduler');
+const JobFabrik = require('./JobFabrik');
 const errors = require('../errors');
 const scheduler = new Scheduler();
 const Repository = require('../db').Repository;
 const logger = require('../logger');
 const CONST = require('../constants');
 
 class ScheduleManager {
-  static schedule(name, jobType, interval, jobData, user) {
+
+  static runAt(name, jobType, runAt, jobData, user) {
     let agendaJob = {};
-    if (interval === CONST.BACKUP.SCHEDULE.DAILY) {
-      interval = this.getRandomDailySchedule();
-      logger.info(`Schedule interval input as 'daily'. So setting following daily random schedule - ${interval}`);
-    }
     return scheduler
-      .schedule(name, jobType, interval, jobData)
+      .runAt(name, jobType, runAt, jobData)
       .tap(job => agendaJob = job)
-      .then(() => this.saveJob(name, jobType, interval, jobData, user))
+      .then(() => this.saveJob(name, jobType, runAt, jobData, user, true))
       .then(jobInDB => this.getJobAttrs(jobInDB, agendaJob));
   }
 
+  static schedule(name, jobType, interval, jobData, user) {
+    return Promise.try(() => {
+      let agendaJob = {};
+      if (interval === CONST.SCHEDULE.DAILY) {
+        interval = this.getRandomDailySchedule();
+        logger.info(`Schedule interval input as 'daily'. So setting following daily random schedule - ${interval}`);
+      } else if (interval === CONST.SCHEDULE.RANDOM) {
```

Comment:
the repeat interval is implemented by 'ServiceInstanceUpdateJob' only. If for other job, let's say for Backup scheduler if user inputs random as repeat interval will it fail ??

Reply:
Please look at [BaseJob](https://github.com/jagadish-kb/service-fabrik-broker/blob/master/lib/jobs/BaseJob.js#L88). Yes, end user will see an error.